packages/core/src/components/overlay/overlay.tsx

```diff
@@ -309,10 +309,15 @@ export class Overlay extends React.Component<IOverlayProps, IOverlayState> {
     }
 
     private handleBackdropMouseDown = (e: React.MouseEvent<HTMLDivElement>) => {
-        if (this.props.canOutsideClickClose) {
-            safeInvoke(this.props.onClose, e);
+        const { backdropProps, canOutsideClickClose, enforceFocus, onClose } = this.props;
+        if (canOutsideClickClose) {
+            safeInvoke(onClose, e);
+        }
+        if (enforceFocus) {
+            // make sure document.activeElement is updated before bringing the focus back
+            requestAnimationFrame(() => this.bringFocusInsideOverlay());
```

Comment:
we've enjoyed success using `rAF` for focus things, more than with `setTimeout` because it executes later _in the same frame_. i support this.